avogadro/rendering/solidpipeline.cpp

```diff
@@ -0,0 +1,132 @@
+/******************************************************************************
+  This source file is part of the Avogadro project.
+  This source code is released under the 3-Clause BSD License, (see "LICENSE").
+******************************************************************************/
+
+#include "solidpipeline.h"
+
+#include "solid_vs.h"
+
+#include "solid_first_fs.h"
+
+#include <iostream>
+
+namespace Avogadro::Rendering {
+
+SolidPipeline::SolidPipeline()
+{
+}
+
+void initializeFramebuffer(GLuint *outFBO, GLuint *texRGB, GLuint *texDepth)
+{
+  glGenFramebuffers(1, outFBO);
+  glBindFramebuffer(GL_FRAMEBUFFER, *outFBO);
+
+  glGenTextures(1, texRGB);
+  glBindTexture(GL_TEXTURE_2D, *texRGB);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, *texRGB, 0);
+
+  glGenTextures(1, texDepth);
+  glBindTexture(GL_TEXTURE_2D, *texDepth);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  glFramebufferTexture2D(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_TEXTURE_2D, *texDepth, 0);
+}
+
+void SolidPipeline::initialize()
+{
+  initializeFramebuffer(&m_renderFBO, &m_renderTexture, &m_depthTexture);
+  resize(1024, 768);
```

Comment:
Do we want this to be adjustable from the widget sizes or for AA?

Reply:
I'm not sure if that line is needed, I might remove it as well...